cfme/tests/automate/custom_button/test_service_objects.py

```diff
@@ -153,3 +157,99 @@ def test_custom_button_display(request, appliance, context, display, objects, bu
+@pytest.mark.uncollectif(
+    lambda context, button_group: context == ViaSSUI and "GENERIC" in button_group
+)
+def test_custom_button_automate(request, appliance, context, submit, objects, button_group):
+    """ Test custom button for automate and requests count as per submit
+
+    prerequisites:
+        * Appliance with service and generic object
+
+    Steps:
+        * Create custom button group with the Object type
+        * Create a custom button with specific submit option and Single and list display
+        * Navigate to object type pages (All and Details)
+        * Check for button group and button
+        * Select/execute button from group dropdown for selected entities
+        * Check for the proper flash message related to button execution
+        * Check automation log requests. Submitted as per selected submit option or not.
+        * Submit all: single request for all entities execution
+        * One by one: separate requests for all entities execution
+
+    Bugzillas:
+        * 1650066
+    """
+
+    group, obj_type = button_group
+    with appliance.context.use(ViaUI):
+        button = group.buttons.create(
+            text=fauxfactory.gen_alphanumeric(),
+            hover=fauxfactory.gen_alphanumeric(),
+            display_for="Single and list",
+            submit=submit,
+            system="Request",
+            request="InspectMe",
+        )
+        request.addfinalizer(button.delete_if_exists)
+
+    with appliance.context.use(context):
+        navigate_to = ssui_nav if context is ViaSSUI else ui_nav
+
+        # BZ-1650066: no custom button on All page
+        destinations = (
+            ["Details"]
+            if context == ViaSSUI and BZ(1650066, forced_streams=["5.9", "5.10"]).blocks
+            else ["All", "Details"]
+        )
+        for destination in destinations:
+            obj = objects[obj_type][destination][0]
+            dest_name = objects[obj_type][destination][1]
+            view = navigate_to(obj, dest_name)
+            custom_button_group = Dropdown(view, group.text)
+            assert custom_button_group.has_item(button.text)
+
+            # Entity count depends on the destination for `All` available entities and
+            # `Details` means a single entity.
+
+            if destination == "All":
+                try:
+                    paginator = view.paginator
+                except AttributeError:
+                    paginator = view.entities.paginator
+
+                entity_count = min(
+                    paginator.items_amount, paginator.items_per_page
+                )
+                view.entities.paginator.check_all()
+            else:
+                entity_count = 1
+
+            # Clear the automation log
+            assert appliance.ssh_client.run_command(
+                'echo -n "" > /var/www/miq/vmdb/log/automation.log'
+            )
+
+            custom_button_group.item_select(button.text)
+            if context != ViaSSUI:
```

Comment:
What happens in SSUI case . Is  there a way to assert in SSUI ?

Reply:
`log request` check is our main check. SSUI not showing msg. I don't think its important as well.

Reply:
I think this is worth putting in as a comment ^

Reply:
@jawatts no need of this... Its expected behavior of SSUI.

Reply:
I think it does need to be in a comment in-line, as its only a _current_ expected behavior, and obviously not every contributor will be aware of this expectation. :)

**Required**

Reply:
I also think this would be better as a positive assertion of the ViaUI context, than negative assertion against ViaSSUI.

Reply:
:D added